wandb/sdk/lib/filesystem.py

```diff
@@ -190,3 +225,33 @@ def safe_copy(source_path: StrPath, target_path: StrPath) -> StrPath:
         shutil.copy2(source_path, tmp_path)
         tmp_path.replace(output_path)
     return target_path
+
+
+def check_exists(path: StrPath) -> Optional[StrPath]:
+    """Look for variations of `path` and return the first found.
+
+    This exists to support former behavior around system-dependent paths; we used to use
+    ':' in Artifact paths unless we were on Windows, but this has issues when e.g. a
+    Linux machine is accessing an NTFS filesystem; we might need to look for the
+    alternate path. This checks all the possible directories we would consider creating.
+    """
+    for dest in path_fallbacks(path):
+        if os.path.exists(dest):
+            return Path(dest) if isinstance(path, Path) else dest
```

Comment:
Why do you need to do the check if it's already Path?

Saw it in multiple places, so same comment applies to the other instances

Reply:
Caller convenience; if they pass in a string, return a string, if they pass in a path, return a path.

We could use `type(path)(dest)` instead which is a little more 'cute', but requires a `#type: ignore` and is maybe less obvious what it's doing.